### ISSUE TITLE
Empty string in table

### ISSUE THREAD

**Issue by dotRData** [rstudio/sparklyr#1802]
This problem is very hard to reproduce, as I don't know, what is the problem.
I have been using sparklyr for last 6 months and this is the first time i am facing a problem I am not able to find anywhere. 

Issue is, I have a table which has string columns data, sparklyr is not able to read few of the columns. I have no idea, which kind of column it can read, which kind of column it can not read

# Connection 
```
config <- spark_config()
config$`sparklyr.shell.driver-cores`         <- 50
config$`sparklyr.shell.driver-memory`        <- "16g"
config$`sparklyr.shell.executor-memory`      <- "16g"
config$`sparklyr.shell.num-executors`        <- 50
config$`spark.driver.maxResultSize`          <- '5g'
config$`spark.driver.memory`                 <- "10g"
config$`spark.dynamicAllocation.enabled`     <- "false"
config$`spark.executor.cores`                <- 5
config$`spark.executor.memory`               <- "16g"
config$`spark.executor.instances`            <- 50
config$`spark.speculation`                   <- "true"
config$`spark.speculation.multiplier`        <- 1.5
config$`spark.speculation.quantile`          <- 0.75
config$`spark.sql.parquet.writeLegacyFormat` <- "true"
config$`spark.yarn.am.cores`                 <- 10
config$`spark.yarn.am.memory`                <- "16g"
config$`spark.yarn.executor.memoryOverhead`  <- "4g"

sc <- spark_connect(master = "yarn-client", spark_home = '/var/rahul/spark-2.1.1', config = config)
```

# SQL Data 
```
desc test_sparklyr;
col_name	data_type	comment
col1                	string
col2                	string
col3                	string
col4                	string
Time taken: 0.049 seconds, Fetched: 4 row(s)

select * from test_sparklyr limit 10;
test_sparklyr.col1	test_sparklyr.col2	test_sparklyr.col3	test_sparklyr.col4
iPhone	cfdf0d92-c62e-11e2-8972-0025906125ac	546845F6-A7B0-7399-0F1A-1D5A02CE587C	7072FC5C-902E-4E51-8892-594A27890375
iPhone	d0950e4c-25f9-11e2-9703-00259069d5fe	2FF75A11-7FEB-7A4F-DB82-B222643206E8	555B9BE0-7F8B-4373-BD32-F3401D4728A1
Android	db59d4a0-5232-11e8-81c1-52547424cb43	0e295af0-645e-3587-89d5-f2bd89310183	238520b5-b0aa-4b99-b264-78637c04f430
Android	3d702106-07db-11e2-9ac1-0025906125f0	1647b0fb-1ac0-3aac-80dc-8d4304ad2255	255a27ff-3bad-4447-90bd-76cc83e98f19
iPhone	31f3ed3c-f4cc-11e3-a18c-002590c12a88	1F28730F-D053-5A0A-D5A1-BDE76B498851	B46C791E-2C05-4EC0-BC9F-0CA00DB65D06
iPhone	29d8d338-23cc-11e2-86db-00259069d5fe	FC11E2E2-FDCB-1849-D556-460D48DC78EA	00000000-0000-0000-0000-000000000000
Android	0ae87b0c-1956-11e2-b77c-0025906124fe	6e72a4cc-8469-3af6-af7a-fc5b4e96c1fd	06e31c50-da46-4e96-a30a-e2f1cb3d6440
iPhone	ce15f274-a521-11e2-a28b-00259060b5bc	81F87085-F026-D316-2872-2ABF4E08972D	00000000-0000-0000-0000-000000000000
Android	11c72786-d2fd-11e3-b05f-0025906704a2	e686661f-0bd0-3422-ba70-3f3f397dfdfa	fd41315a-be6f-4beb-812e-31f587077e9a
Android	76838ab8-289d-11e2-8ce8-00259069d5fe	7c65b18e-25b4-3a5a-896a-8b825341bf10	d2d28c7d-3992-404a-81ed-1978a689557d

select 
count(distinct(col1)),
count(distinct(col2)),
count(distinct(col3)),
count(distinct(col4)) 
from test_sparklyr ;

c0	c1	c2	c3
3	583723	583328	514571
```

# Data from sparklyr
```
sc %>% tbl('test_sparklyr') %>% head(10) %>% collect %>% View
```
<img width="754" alt="screen shot 2018-12-10 at 10 21 15 pm" src="https://user-images.githubusercontent.com/4815729/49748267-3c001880-fccb-11e8-9875-e65474c9f08d.png">

```
> sc %>% tbl('test_sparklyr') %>% summarise(n_distinct(col1),n_distinct(col2),n_distinct(col3),n_distinct(col4))
# Source: spark<?> [?? x 4]
  `n_distinct(col1)` `n_distinct(col2)` `n_distinct(col3)` `n_distinct(col4)`
*              <dbl>              <dbl>              <dbl>              <dbl>
1                  1                  1             583328             514571
```

# session Info
```
> devtools::session_info()
Session info -----------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.3 (2017-11-30)
 system   x86_64, linux-gnu           
 ui       RStudio (1.2.1015)          
 language (EN)                        
 collate  en_US.UTF-8                 
 tz       <NA>                        
 date     2018-12-10                  

Packages ---------------------------------------------------------------------------------
 package       * version     date       source                               
 assertthat      0.2.0       2017-04-11 CRAN (R 3.4.3)                       
 backports       1.1.2       2017-12-13 CRAN (R 3.4.3)                       
 base          * 3.4.3       2018-01-10 local                                
 base64enc       0.1-3       2015-07-28 CRAN (R 3.4.3)                       
 broom           0.5.1.9000  2018-12-10 Github (tidyverse/broom@9a85f75)     
 cellranger      1.1.0       2016-07-27 CRAN (R 3.4.3)                       
 cli             1.0.0       2017-11-05 CRAN (R 3.4.3)                       
 colorspace      1.3-2       2016-12-14 CRAN (R 3.4.3)                       
 compiler        3.4.3       2018-01-10 local                                
 config          0.2         2016-08-02 CRAN (R 3.4.3)                       
 crayon          1.3.4       2017-09-16 CRAN (R 3.4.3)                       
 datasets      * 3.4.3       2018-01-10 local                                
 DBI           * 1.0.0       2018-05-02 Github (cran/DBI@e390c64)            
 dbplyr          1.2.1.9001  2018-10-04 Github (tidyverse/dbplyr@ce9273e)    
 devtools        1.13.4      2017-11-09 CRAN (R 3.4.3)                       
 digest          0.6.13      2017-12-14 CRAN (R 3.4.3)                       
 dplyr         * 0.7.99.9000 2018-12-10 Github (tidyverse/dplyr@6e17633)     
 forcats       * 0.3.0.9000  2018-10-04 Github (tidyverse/forcats@0edac19)   
 forge           0.1.0       2018-08-31 Github (cran/forge@c8d7bf6)          
 generics        0.0.2       2018-12-10 Github (r-lib/generics@d59e6b4)      
 ggplot2       * 3.1.0.9000  2018-12-10 Github (tidyverse/ggplot2@4a607ca)   
 glue            1.3.0       2018-12-10 Github (tidyverse/glue@35c61e9)      
 graphics      * 3.4.3       2018-01-10 local                                
 grDevices     * 3.4.3       2018-01-10 local                                
 grid            3.4.3       2018-01-10 local                                
 gtable          0.2.0       2016-02-26 CRAN (R 3.4.3)                       
 haven           2.0.0.9000  2018-12-10 Github (hadley/haven@f0dc4e5)        
 hms             0.4.2.9001  2018-10-04 Github (tidyverse/hms@979286f)       
 htmltools       0.3.6       2017-04-28 CRAN (R 3.4.3)                       
 htmlwidgets     1.3         2018-12-10 Github (ramnathv/htmlwidgets@5db27ad)
 httpuv          1.4.5.9001  2018-12-10 Github (rstudio/httpuv@a830a0d)      
 httr            1.3.1       2017-08-20 CRAN (R 3.4.3)                       
 jsonlite        1.5         2017-06-01 CRAN (R 3.4.3)                       
 later           0.7.5.9000  2018-12-10 Github (r-lib/later@2318607)         
 lazyeval        0.2.1       2017-10-29 CRAN (R 3.4.3)                       
 lubridate       1.7.4.9000  2018-12-10 Github (tidyverse/lubridate@22f35c2) 
 magrittr      * 1.5         2014-11-22 CRAN (R 3.4.3)                       
 memoise         1.1.0       2017-04-21 CRAN (R 3.4.3)                       
 methods       * 3.4.3       2018-01-10 local                                
 mime            0.5         2016-07-07 CRAN (R 3.4.3)                       
 modelr          0.1.1       2017-07-24 CRAN (R 3.4.3)                       
 munsell         0.4.3       2016-02-13 CRAN (R 3.4.3)                       
 openssl         0.9.9       2017-11-10 CRAN (R 3.4.3)                       
 parallel        3.4.3       2018-01-10 local                                
 pillar          1.3.0       2018-07-14 Github (cran/pillar@aab68fa)         
 pkgconfig       2.0.1       2017-03-21 CRAN (R 3.4.3)                       
 plyr            1.8.4       2016-06-08 CRAN (R 3.4.3)                       
 promises        1.0.1.9000  2018-12-10 Github (rstudio/promises@852e9be)    
 purrr         * 0.2.4       2017-10-18 CRAN (R 3.4.3)                       
 r2d3            0.2.2       2018-10-07 Github (rstudio/r2d3@c121ec8)        
 R6              2.2.2       2017-06-17 CRAN (R 3.3.2)                       
 Rcpp            1.0.0.1     2018-12-10 Github (RcppCore/Rcpp@0664c1a)       
 readr         * 1.1.1       2017-05-16 CRAN (R 3.4.3)                       
 readxl          1.0.0       2017-04-18 CRAN (R 3.4.3)                       
 RevoUtils     * 10.0.8      2018-02-20 local                                
 RevoUtilsMath * 10.0.1      2017-09-19 local                                
 rlang           0.3.0.9001  2018-12-10 Github (r-lib/rlang@ce9a1a6)         
 rprojroot       1.3-1       2017-12-18 CRAN (R 3.4.3)                       
 rstudioapi      0.7         2017-09-07 CRAN (R 3.4.3)                       
 rvest           0.3.2       2016-06-17 CRAN (R 3.4.3)                       
 scales          0.5.0       2017-08-24 CRAN (R 3.4.3)                       
 shiny           1.2.0.9000  2018-12-10 Github (rstudio/shiny@6a373b5)       
 sparklyr      * 0.9.3.9000  2018-12-10 Github (rstudio/sparklyr@f20fa1d)    
 stats         * 3.4.3       2018-01-10 local                                
 stringi         1.2.4       2018-07-20 Github (cran/stringi@9d88590)        
 stringr       * 1.3.1.9000  2018-10-04 Github (tidyverse/stringr@0b5a4c4)   
 tibble        * 1.4.2       2018-01-22 Github (cran/tibble@d621a51)         
 tidyr         * 0.8.2.9000  2018-12-10 Github (tidyverse/tidyr@28af2f1)     
 tidyselect      0.2.5.9000  2018-12-10 Github (tidyverse/tidyselect@48eb3a6)
 tidyverse     * 1.2.1.9000  2018-10-04 Github (hadley/tidyverse@66e8bbd)    
 tools           3.4.3       2018-01-10 local                                
 utils         * 3.4.3       2018-01-10 local                                
 withr           2.1.1       2017-12-19 CRAN (R 3.4.3)                       
 xml2            1.1.1       2017-01-24 CRAN (R 3.4.3)                       
 xtable          1.8-2       2016-02-05 CRAN (R 3.4.3)                       
 yaml            2.1.16      2017-12-12 CRAN (R 3.4.3) 

> utils::sessionInfo()
R version 3.4.3 (2017-11-30)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS release 6.5 (Final)

Matrix products: default
BLAS: /var/rahul/anaconda3/lib/R/lib/libRblas.so
LAPACK: /var/rahul/anaconda3/lib/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] magrittr_1.5         DBI_1.0.0            sparklyr_0.9.3.9000  forcats_0.3.0.9000  
 [5] stringr_1.3.1.9000   dplyr_0.7.99.9000    purrr_0.2.4          readr_1.1.1         
 [9] tidyr_0.8.2.9000     tibble_1.4.2         ggplot2_3.1.0.9000   tidyverse_1.2.1.9000
[13] RevoUtils_10.0.8     RevoUtilsMath_10.0.1

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5.9000 forge_0.1.0           haven_2.0.0.9000     
 [4] colorspace_1.3-2      generics_0.0.2        htmltools_0.3.6      
 [7] yaml_2.1.16           base64enc_0.1-3       rlang_0.3.0.9001     
[10] pillar_1.3.0          later_0.7.5.9000      glue_1.3.0           
[13] withr_2.1.1           dbplyr_1.2.1.9001     modelr_0.1.1         
[16] readxl_1.0.0          plyr_1.8.4            munsell_0.4.3        
[19] gtable_0.2.0          cellranger_1.1.0      rvest_0.3.2          
[22] devtools_1.13.4       htmlwidgets_1.3       memoise_1.1.0        
[25] httpuv_1.4.5.9001     parallel_3.4.3        broom_0.5.1.9000     
[28] r2d3_0.2.2            Rcpp_1.0.0.1          xtable_1.8-2         
[31] openssl_0.9.9         scales_0.5.0          backports_1.1.2      
[34] promises_1.0.1.9000   jsonlite_1.5          config_0.2           
[37] mime_0.5              hms_0.4.2.9001        digest_0.6.13        
[40] stringi_1.2.4         shiny_1.2.0.9000      grid_3.4.3           
[43] rprojroot_1.3-1       cli_1.0.0             tools_3.4.3          
[46] lazyeval_0.2.1        crayon_1.3.4          pkgconfig_2.0.1      
[49] xml2_1.1.1            lubridate_1.7.4.9000  assertthat_0.2.0     
[52] httr_1.3.1            rstudioapi_0.7        R6_2.2.2             
[55] compiler_3.4.3  
```

**Comment by kevinykuo** [comment]
What happens if you do `sc %>% tbl('test_sparklyr') %>% summarise(n_distinct(col1),n_distinct(col2),n_distinct(col3),n_distinct(col4)) %>% dbplyr::sql_render()` and execute the resulting statement in the shell?

**Comment by dotRData** [comment]
Same issue. 

# R
```r
sc %>% tbl('test_sparklyr') %>% summarise(n_distinct(col1),n_distinct(col2),n_distinct(col3),n_distinct(col4)) %>% dbplyr::sql_render()

<SQL> SELECT count(DISTINCT(`col1`)) AS `n_distinct(col1)`, count(DISTINCT(`col2`)) AS `n_distinct(col2)`, count(DISTINCT(`col3`)) AS `n_distinct(col3)`, count(DISTINCT(`col4`)) AS `n_distinct(col4)`
FROM `test_sparklyr`
```

# HIVE-CLI
```
SELECT count(DISTINCT(`col1`)) AS `n_distinct(col1)`, count(DISTINCT(`col2`)) AS `n_distinct(col2)`, count(DISTINCT(`col3`)) AS `n_distinct(col3)`, count(DISTINCT(`col4`)) AS `n_distinct(col4)` FROM `test_sparklyr`
OK
n_distinct(col1)	n_distinct(col2)	n_distinct(col3)	n_distinct(col4)
3	583723	583328	514571
```

**Comment by dotRData** [comment]
@kevinykuo .. an idea what would be happening here .. i tried reinstalling package .. but still the same thing ...

**Comment by dotRData** [comment]
this has also happened in the past .. when the column type was "array<string>", but it was not that important ... so i let it go. but not reading a string column was unexpected. 

```
select col1 from  test_table  where size(col1) > 0  limit 10;

col1
["eee85d9f-2dfb-445a-8176-4d4ab6f722e1"]
["1098341183","1098329255","1098340117"]
["1098185983","1098187479"]
["1098195641"]
["1098826441"]
["6349a100-b05f-4456-a836-5e10245eee02"]
["e37e41d2-b3ff-4a4b-bd26-53d9a4b034f6"]
["136ea5cc-ad43-4d95-a002-203045edde26","ec1db5a2-0837-4dbf-87e4-7dcdf936cd12"]
["d8fad8c6-3806-4437-bed0-8dc9336f1a33","8dde6e40-27ba-440f-b8e6-508156ff7b77"]
["e2954bfa-69b5-4d6f-9e03-e0073efe9cce"]
```

```
> sc %>% 
+   tbl('test_table') %>% 
+   filter(size(col1) > 0) %>% 
+   select(col1)

# Source: spark<?> [?? x 1]
# ... with 1 variable: col1 <lgl>

```

**Comment by kevinykuo** [comment]
This is definitely tough. Could you execute the query through spark-shell via Spark SQL instead of through Hive directly? I want to first make sure it's a sparklyr specific issue and not a Spark issue.

**Comment by dotRData** [comment]
its working on spark-shell 

```
import org.apache.spark.sql.hive.HiveContext
val hiveContext = new org.apache.spark.sql.hive.HiveContext(sc)
import hiveContext._
```

## Sample Data
```
hiveContext.sql("SELECT * FROM  test_sparklyr limit 10").collect().foreach(println)

[Android,2b468d9d-6e8d-43ff-a846-e8e8ba348712,7a6ee8d1-2c19-3c5a-890c-0ddc88baca69,d36cc264-e50e-49b9-bfbe-5c29cb224c38]
[Android,0717de34-108b-4b0e-baa8-154bdc8db5da,1ff4f605-15e6-3641-a265-796e7b0fbb00,5d69c772-63ea-4f92-ad2c-f097cca337bb]
[Android,3cb6fe50-2411-4342-be38-6a7e9d66f4d0,f2034044-5460-323e-bdc3-39031a84dae3,01cacfd4-4e54-4690-8e93-8e6d9e31c96e]
[Android,0717de34-108b-4b0e-baa8-154bdc8db5da,1ff4f605-15e6-3641-a265-796e7b0fbb00,5d69c772-63ea-4f92-ad2c-f097cca337bb]
[Android,3cb6fe50-2411-4342-be38-6a7e9d66f4d0,f2034044-5460-323e-bdc3-39031a84dae3,01cacfd4-4e54-4690-8e93-8e6d9e31c96e]
[Android,a8a1d2a1-2581-4b33-949b-7b3530f8ac6b,826b9969-2374-38a8-8ebe-645c8a8f01df,00000000-0000-0000-0000-000000000000]
[Android,57f8f09a-26f1-11e2-8757-00259069d5fe,d80746b0-91d4-3372-9783-7776ac591514,38431066-a761-4abc-be74-3ca0b949cf0f]
[Android,57f8f09a-26f1-11e2-8757-00259069d5fe,d80746b0-91d4-3372-9783-7776ac591514,38431066-a761-4abc-be74-3ca0b949cf0f]
[Android,a8a1d2a1-2581-4b33-949b-7b3530f8ac6b,826b9969-2374-38a8-8ebe-645c8a8f01df,00000000-0000-0000-0000-000000000000]
[Android,a8a1d2a1-2581-4b33-949b-7b3530f8ac6b,826b9969-2374-38a8-8ebe-645c8a8f01df,00000000-0000-0000-0000-000000000000]
```

## Distinct Value
```
hiveContext.sql("SELECT count(DISTINCT(`col1`)) AS `n_distinct(col1)`, count(DISTINCT(`col2`)) AS `n_distinct(col2)`, count(DISTINCT(`col3`)) AS `n_distinct(col3)`, count(DISTINCT(`col4`)) AS `n_distinct(col4)` FROM `test_sparklyr`").collect().foreach(println)

[3,583723,583328,514571]
```

## For array
```
hiveContext.sql("select col1 from  test_table  where size(col1) > 0  limit 10").collect().foreach(println)

[WrappedArray(eee85d9f-2dfb-445a-8176-4d4ab6f722e1)]
[WrappedArray(1098341183, 1098329255, 1098340117)]
[WrappedArray(1098185983, 1098187479)]
[WrappedArray(1098195641)]
[WrappedArray(1098826441)]
[WrappedArray(6349a100-b05f-4456-a836-5e10245eee02)]
[WrappedArray(e37e41d2-b3ff-4a4b-bd26-53d9a4b034f6)]
[WrappedArray(136ea5cc-ad43-4d95-a002-203045edde26, ec1db5a2-0837-4dbf-87e4-7dcdf936cd12)]
[WrappedArray(d8fad8c6-3806-4437-bed0-8dc9336f1a33, 8dde6e40-27ba-440f-b8e6-508156ff7b77)]
[WrappedArray(e2954bfa-69b5-4d6f-9e03-e0073efe9cce)]
```

**Comment by kevinykuo** [comment]
Could you try something like this

```
spark_session(sc) %>%
  invoke("sql", "[INSERT STATEMENT HERE]") %>%
  invoke("collect")
```

Does it return anything for the problematic column?

**Comment by dotRData** [comment]
same issue ..

```
> spark_session(sc) %>%
+   sparklyr::invoke("sql", "SELECT count(DISTINCT(col1)), count(DISTINCT(col2)), count(DISTINCT(col3)), count(DISTINCT(col4)) FROM test_sparklyr") %>%
+   sparklyr::invoke("collect")
[[1]]
[[1]][[1]]
[1] 1

[[1]][[2]]
[1] 1

[[1]][[3]]
[1] 583328

[[1]][[4]]
[1] 514571

```


```
> spark_session(sc) %>%
+   sparklyr::invoke("sql", "SELECT * FROM test_sparklyr limit 2") %>%
+   sparklyr::invoke("collect")
[[1]]
[[1]][[1]]
[1] ""

[[1]][[2]]
[1] ""

[[1]][[3]]
[1] "52D1100A-6D7C-C0EB-B098-5A10FCAD55D8"

[[1]][[4]]
[1] "28A8382A-1597-48F9-8A68-958C984CADF6"


[[2]]
[[2]][[1]]
[1] ""

[[2]][[2]]
[1] ""

[[2]][[3]]
[1] "7FDE78BA-28EF-C402-0A87-3AC25FC596A5"

[[2]][[4]]
[1] "9E22DAC7-59F4-45A0-A320-8B1BD95BD0EF"

```

**Comment by kevinykuo** [comment]
So `invoke()` literally executes scala code in the backend so I'm surprised to see we're getting a different result. One last thing to try would be to change the connection code to `sc <- spark_connect(master = "yarn-client", config = config)` and seeing if that works (or even connects) to rule out issues with incorrect metastores. After that, I'd probably try to identify the rows in that specific column that's giving us trouble.

**Comment by dotRData** [comment]
without spark_home, it doesn't connect.

one more issue I have seen is, spark-2.1.1 connect to the cluster but spark-2.3.1 doesn't. 
It works in spark-shell but not via sparklyr.

```
> sc <- spark_connect(master = "yarn-client")
Error in shell_connection(master = master, spark_home = spark_home, app_name = app_name,  : 
  Failed to connect to Spark (SPARK_HOME is not set).

> sc <- spark_connect(master = "yarn-client", spark_home = '/var/rahul/spark-2.1.1')
> spark_session(sc) %>%
+     sparklyr::invoke("sql", "SELECT count(DISTINCT(col1)), count(DISTINCT(col2)), count(DISTINCT(col3)), count(DISTINCT(col4)) FROM default.test_sparklyr") %>%
+     sparklyr::invoke("collect")
[[1]]
[[1]][[1]]
[1] 1

[[1]][[2]]
[1] 1

[[1]][[3]]
[1] 583328

[[1]][[4]]
[1] 514571


> spark_disconnect_all()
[1] 1
> sc <- spark_connect(master = "yarn-client", spark_home = '/var/rahul/spark-2.3.1')
> spark_session(sc) %>%
+   sparklyr::invoke("sql", "SELECT count(DISTINCT(col1)), count(DISTINCT(col2)), count(DISTINCT(col3)), count(DISTINCT(col4)) FROM default.test_sparklyr") %>%
+   sparklyr::invoke("collect")
Error: org.apache.spark.sql.AnalysisException: Table or view not found: `default`.`test_sparklyr`; line 1 pos 103;
'Project [unresolvedalias('count('col1), None), unresolvedalias('count('col2), None), unresolvedalias('count('col3), None), unresolvedalias('count('col4), None)]
+- 'UnresolvedRelation `default`.`test_sparklyr`

	at org.apache.spark.sql.catalyst.analysis.package$AnalysisErrorAt.failAnalysis(package.scala:42)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$$anonfun$checkAnalysis$1.apply(CheckAnalysis.scala:82)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$$anonfun$checkAnalysis$1.apply(CheckAnalysis.scala:80)
	at org.apache.spark.sql.catalyst.trees.TreeNode.foreachUp(TreeNode.scala:127)
	at org.apache.spark.sql.catalyst.trees.TreeNode$$anonfun$foreachUp$1.apply(TreeNode.scala:126)
	at org.apache.spark.sql.catalyst.trees.TreeNode$$anonfun$foreachUp$1.apply(TreeNode.scala:126)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at org.apache.spark.sql.catalyst.trees.TreeNode.foreachUp(TreeNode.scala:126)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$class.checkAnalysis(CheckAnalysis.scala:80)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.checkAnalysis(Analyzer.scala:92)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.executeAndCheck(Analyzer.scala:105)
	at org.apache.spark.sql.execution.QueryExecution.analyzed$lzycompute(QueryExecution.scala:57)
	at org.apache.spark.sql.execution.QueryExecution.analyzed(QueryExecution.scala:55)
	at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:47)
	at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:74)
	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:641)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at sparklyr.Invoke.invoke(invoke.scala:139)
	at sparklyr.StreamHandler.handleMethodCall(stream.scala:123)
	at sparklyr.StreamHandler.read(stream.scala:66)
	at sparklyr.BackendHandler.channelRead0(handler.scala:51)
	at sparklyr.BackendHandler.channelRead0(handler.scala:4)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:284)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1359)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:935)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:138)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:645)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by dotRData** [comment]
hope this helps.

```
sc
$master
[1] "yarn-client"

$method
[1] "shell"

$app_name
[1] "sparklyr"

$config
$config$spark.env.SPARK_LOCAL_IP.local
[1] "127.0.0.1"

$config$sparklyr.connect.csv.embedded
[1] "^1.*"

$config$spark.sql.catalogImplementation
[1] "hive"

$config$sparklyr.connect.cores.local
[1] 24

$config$spark.sql.shuffle.partitions.local
[1] 24

$config$sparklyr.gateway.port
[1] 8884

$config$`sparklyr.shell.driver-cores`
[1] 50

$config$`sparklyr.shell.driver-memory`
[1] "16g"

$config$`sparklyr.shell.executor-memory`
[1] "16g"

$config$`sparklyr.shell.num-executors`
[1] 50

$config$spark.driver.maxResultSize
[1] "5g"

$config$spark.driver.memory
[1] "10g"

$config$spark.dynamicAllocation.enabled
[1] "false"

$config$spark.executor.cores
[1] 5

$config$spark.executor.memory
[1] "16g"

$config$spark.executor.instances
[1] 50

$config$spark.speculation
[1] "true"

$config$spark.speculation.multiplier
[1] 1.5

$config$spark.speculation.quantile
[1] 0.75

$config$spark.sql.parquet.writeLegacyFormat
[1] "true"

$config$spark.yarn.am.cores
[1] 10

$config$spark.yarn.am.memory
[1] "16g"

$config$spark.yarn.executor.memoryOverhead
[1] "4g"

$config$spark.yarn.queue
[1] "public"

attr(,"config")
[1] "default"
attr(,"file")
[1] "/home/rahul/anaconda3/lib/R/library/sparklyr/conf/config-template.yml"

$state
<environment: 0x7f14dd663a58>

$spark_home
[1] "/var/rahul/spark-2.1.1"

$backend
A connection with                              
description "->localhost:8885"
class       "sockconn"        
mode        "wb"              
text        "binary"          
opened      "opened"          
can read    "yes"             
can write   "yes"             

$monitoring
A connection with                              
description "->localhost:8885"
class       "sockconn"        
mode        "wb"              
text        "binary"          
opened      "opened"          
can read    "yes"             
can write   "yes"             

$gateway
A connection with                              
description "->localhost:8884"
class       "sockconn"        
mode        "rb"              
text        "binary"          
opened      "opened"          
can read    "yes"             
can write   "yes"             

$output_file
[1] "/tmp/RtmpFaUPtR/file18da28df6ee8_spark.log"

$sessionId
[1] 16017

attr(,"class")
[1] "spark_connection"       "spark_shell_connection" "DBIConnection"     
```

**Comment by dotRData** [comment]
even this is not working 

```
> sc %>% dbGetQuery("select * from test_sparklyr limit 10")
   col1 col2                                 col3                                 col4
1            52D1100A-6D7C-C0EB-B098-5A10FCAD55D8 28A8382A-1597-48F9-8A68-958C984CADF6
2            7FDE78BA-28EF-C402-0A87-3AC25FC596A5 9E22DAC7-59F4-45A0-A320-8B1BD95BD0EF
3            a2fe5c21-c33e-322e-905d-7d4ec5c5a8e6 6a0ce742-6b33-4b22-8995-0382eed94062
4            80b9ab95-d102-342d-86e9-121fc8372652 9248c5dc-25e7-42ae-bde9-cc95f438f5d0
5            DBDA41A9-949A-30BF-F5C7-3483662DE0AF E797544D-50A3-44A6-9AB6-88CC2142D9F5
6            A6A74D6C-D2C9-921B-7DF0-1A67C1B10840 00000000-0000-0000-0000-000000000000
7            e4158b22-5992-351e-acf6-637bb681aa3d 9a663b86-f637-40aa-b17e-0301a35ad0af
8            3c6f8e29-20fc-341a-a667-6f9470d201fc 618cf17e-144b-4654-ae5b-16272288fcae
9            fa03dc0f-2cc9-39d3-ae58-09d645ada8fe ea0da550-46da-47d7-9b78-efe736693391
10           A52E5376-5B9F-14B4-598D-6CC195C76DEA 9846614C-9249-45D5-A8A4-AA5A785EB190
```

**Comment by kevinykuo** [comment]
Okay I'll need to find a day to spin up some clusters to look into this and a few other hive related issues.

Could you actually grab a small subset of `test_sparklyr` (like 10 rows), put it into a separate table (via hive cli or something), and see if you can reproduce the issue from sparklyr? I'm suspecting there's something special about the content in those columns. My previous suggestion was to do a binary search and try to narrow it down, but maybe we'll get lucky.

**Comment by dotRData** [comment]
This issue got resolved, once I moved my code to spark-2.3.1.
So closing this issue from my end.
